packages/grafana-runtime/src/services/ScopesDashboardsService.ts

```diff
@@ -0,0 +1,259 @@
+import { isEqual } from 'lodash';
+import { BehaviorSubject } from 'rxjs';
+
+import { ScopeDashboardBinding, SuggestedDashboardsFoldersMap } from '@grafana/data';
+
+import { config } from '../config';
+
+import { getBackendSrv } from './backendSrv';
+
+export interface ScopesDashboardsServiceState {
+  // by keeping a track of the raw response, it's much easier to check if we got any dashboards for the currently selected scopes
+  dashboards: ScopeDashboardBinding[];
+  // a filtered version of the `folders` property. this prevents a lot of unnecessary parsings in React renders
+  filteredFolders: SuggestedDashboardsFoldersMap;
+  // this is a grouping in folders of the `dashboards` property. it is used for filtering the dashboards and folders when the search query changes
+  folders: SuggestedDashboardsFoldersMap;
+  forScopeNames: string[];
+  isEnabled: boolean;
+  isLoading: boolean;
+  isOpened: boolean;
+  isReadOnly: boolean;
+  searchQuery: string;
```

Comment:
I understand the dashboard list pane need some of this state, but is all this state needed from plugins?